src/_main_/utils/common.py

```diff
@@ -106,11 +106,14 @@ def rename_fields(args, pairs):
   return args
 
 
-def serialize_all(data, full=False):
+def serialize_all(data, full=False, **kwargs):
+  medium = (kwargs or {}).get("medium", False)
```

Comment:
This seems kludgy.  Just so you can pass the filter flags back for the community.  It might make sense if there was a medium_json for all the models, but that is not the case.  Can you add the filter flags into community.simple_json which I think will be plenty fast given there aren't and will never be too many filter flags.

Reply:
It isnt actually @BradHN1  🤣 . Let me share what I was thinking and after, if you still think it's "kludgy" we will find a better way 🤣 🤣. 

So there are two parts to this. 

Just like you, I initially thought that I could just add the flags to the simple_json function. I didnt do it because, the flags will now make "simple_json" very heavy. Not in terms of the number of items in the object it returns, but in terms of how long it takes to actually finish running the "simple_json". You notice that when we are trying to get the flags enabled for a community or a user, we actually do a DB fetch for all flags, then do a bunch of other comparisons in loops before we arrive at the flags that are activated for a particular community/user. These extra milliseconds **will** pile up when are calling long lists of communities/users (which we already do in so many places). All of a sudden, "fast light weight list of communities" wont be showing up as quickly anymore. Following this, the only other option I had,  was to call "full_json" during serialization for all the "communities.listFor...." routes, which would have also given us too many unnecessary information for reach community (that would also waste too much time when loading the whole list of communities).

And that is why I just introduced `medium_json`. Doing so, I had to modify our `serialize_all` function. Here is one thing to note, I didnt only modify the structure of the function to forcefully fit in "`medium_json`". You will see that the third argument, isnt specifically "`medium`" it is `**kwargs`. Which now gives us space to pass in anything we want, as many things as we want, and also allow us to modify the `serialize` function however we want, for any special cases that arise without having to track back to it's old uses and editing them to match the new structure. 
So in short, this is actually the last time we would have to change structure of `serialize_all`. Yet, we now have the ability to tweak anything in the function to suit any nuanced cases we meet in future. 

So you see with this new change, any calls to 
`serialize_all(objects,False)` is valid. 
`serialize_all(objects, False, medium=True)` is also valid -- and in future 
`serialize_all(objects, False, something_new=Flase, some_other_thing=[1,2,3], some_other_thing_again=True)` will also be valid.

And all this solves the problem of keeping `simple_json` light weight as intended,  keeping `full_json` for when we need everything, and still catering for the weird case of us needing only lightweight content for all communities, but with their feature flags.

Reply:
For now I will deploy with the feature flags returned with simple_json.  this will not slow anything measurably.